src/muz/spacer/spacer_unsat_core_plugin.cpp

```diff
@@ -134,34 +133,35 @@ namespace spacer {
              * as workaround we take the absolute value of the provided coefficients.
              */
             parameter const* params = d->get_parameters() + 2; // point to the first Farkas coefficient
-            
-            STRACE("spacer.farkas",
-                   verbose_stream() << "Farkas input: "<< "\n";
-                   for (unsigned i = 0; i < m.get_num_parents(step); ++i) {
-                       proof * prem = m.get_parent(step, i);                       
-                       rational coef = params[i].get_rational();                       
-                       bool b_pure = m_learner.m_pr.is_b_pure (prem);
-                       verbose_stream() << (b_pure?"B":"A") << " " << coef << " " << mk_pp(m.get_fact(prem), m) << "\n";
-                   }
-                   );
-            
-            bool can_be_closed = true;
-            
+
+            TRACE("spacer.farkas",
+                  tout << "Farkas input: "<< "\n";
+                  for (unsigned i = 0; i < m.get_num_parents(step); ++i) {
+                      proof * prem = m.get_parent(step, i);
+                      rational coef = params[i].get_rational();
+                      bool b_pure = m_ctx.is_b_pure (prem);
+                      tout << (b_pure?"B":"A") << " " << coef << " " << mk_pp(m.get_fact(prem), m) << "\n";
+                  }
+                );
+
+            bool done = true;
+
```

Comment:
there is an iterator for that. It is called get_parents(step), so you say
for (proof * premise : get_parents(step)) {…. }

Reply:
oh, well, unless you need to access params[i]